core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -1150,7 +1151,17 @@ object SnappyContext extends Logging {
       // clear current hive catalog connection
       Hive.closeCurrent()
       if (ExternalStoreUtils.isLocalMode(sc)) {
-        ServiceUtils.invokeStopFabricServer(sc)
+        val user = sc.getConf.get(Constant.STORE_PROPERTY_PREFIX + Attribute.USERNAME_ATTR, "")
+        val props = if (!user.isEmpty) {
+          val prps = new java.util.Properties();
+          val pass = sc.getConf.get(Constant.STORE_PROPERTY_PREFIX + Attribute.PASSWORD_ATTR, "")
+          prps.put(com.pivotal.gemfirexd.Attribute.USERNAME_ATTR, user)
```

Comment:
sc.getConf will create a copy of SparkConf in every call. Use sc.conf instead which has package-level visibility (and is visible here).